client/my-sites/themes/theme-options.js

```diff
@@ -203,6 +203,41 @@ function getAllThemeOptions( { translate, isFSEActive } ) {
 			isPremiumThemeAvailable( state, themeId, siteId ),
 	};
 
+	// WPCOM-specific plan upgrade for community themes.
+	const upgradePlanForDotOrgThemes = {
+		label: translate( 'Upgrade to activate', {
+			comment: 'label prompting user to upgrade the WordPress.com plan to activate a certain theme',
+		} ),
+		extendedLabel: translate( 'Upgrade to activate', {
+			comment: 'label prompting user to upgrade the WordPress.com plan to activate a certain theme',
+		} ),
+		header: translate( 'Upgrade on:', {
+			context: 'verb',
+			comment: 'label for selecting a site for which to upgrade a plan',
+		} ),
+		getUrl: ( state, themeId, siteId ) => {
+			const { origin = 'https://wordpress.com' } =
+				typeof window !== 'undefined' ? window.location : {};
+			const slug = getSiteSlug( state, siteId );
+
+			const redirectTo = encodeURIComponent(
+				`${ origin }/marketplace/theme/${ themeId }/install/${ slug }`
+			);
+
+			const planPathSlug = getPlanPathSlugForFirstPartyThemes( state, siteId, PLAN_BUSINESS );
+
+			return `/checkout/${ slug }/${ planPathSlug }?redirect_to=${ redirectTo }`;
+		},
+		hideForTheme: ( state, themeId, siteId ) =>
+			isJetpackSite( state, siteId ) ||
+			isSiteWpcomAtomic( state, siteId ) ||
+			! isUserLoggedIn( state ) ||
+			! siteId ||
+			isExternallyManagedTheme( state, themeId ) ||
+			isThemeActive( state, themeId, siteId ) ||
+			isPremiumThemeAvailable( state, themeId, siteId ),
```

Comment:
Im looking at surrounding options and bringing it what I think makes sense here. Can you think of anything else to add or that shouldn't be here?

Reply:
`isJetpackSite` also returns true for atomic sites. 

It appears that ecommerce trial sites are atomic and shouldn't be able to install themes (fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sjcpbz%2Qsrngherf%2Spynff%2Qjcpbz%2Qsrngherf.cuc%3Se%3Q228736oq%23750-og) though I noticed earlier that this seems to be broken, I'll file an issue in the morning.

Maybe adding `!isWporgTheme` too would be safer?

Reply:
On second thought im thinking `isPremiumThemeAvailable` could cause problems here as well? Im thinking I will remove that and add the isWporgTheme check.

Reply:
My first thought here was maybe we should check if the user is an administrator, but I'm a little confused by what `hideForTheme` does. I assumed it would hide the "Upgrade to activate" button, but does it?

Reply:
Thats a good question. It doesn't seem to be used by this page. But by the themes-selection `wrappedPreviewAction`'s secondary action which gets hidden if this is true. But that secondary action is also defined as `null` in many cases as well. Im not sure a case where I actually see a secondary action on the wrapped preview.

Reply:
> My first thought here was maybe we should check if the user is an administrator

forgot to answer this part. But non-admins don't have access to the themes for a site and they are those items are removed from the menu. I tested adding a non admin and that is all removed, i tested visiting directly by URL and while I can browse the showcase, the actions are inert. So I don't think the admin check is necessary, nor used for the other theme options here.

Reply:
The options used to be visible in the theme "more" menu and so `hideForTheme` was very important. We (well @copons) [removed nearly all uses of that menu](https://github.com/Automattic/wp-calypso/pull/85315) with the eventual benefit of removing / massively simplifying this `theme-options.js` file but time caught up with us.

The options are still passed to that component after being [filtered via `hideForTheme`](https://github.com/Automattic/wp-calypso/blob/ff94a7516d50064fda3323f2018bd82a60489aa4/client/my-sites/themes/theme-showcase.jsx#L579), however there's [kind of an allow list there now](https://github.com/Automattic/wp-calypso/blob/ff94a7516d50064fda3323f2018bd82a60489aa4/client/components/theme/index.jsx#L295).